service/src/main/java/com/epam/esm/service/impl/TagServiceImpl.java

```diff
@@ -50,9 +51,12 @@ public void remove(Long id) {
     public TagDTO getMostPopularTag(Long userId) {
         User user = userRepository.getById(userId)
                 .orElseThrow(() -> new EntityNotFoundException(" (id = " + userId + ")"));
-        Tag tag = tagRepository.getMostPopularTag(user.getId())
-                .orElseThrow(EntityNotFoundException::new);
-        return converter.convertToDto(tag);
+        Optional<Tag> tag = tagRepository.getMostPopularTag(user.getId());
+        if(tag.isPresent()){
+            return converter.convertToDto(tag.get());
+        }else {
+            return TagDTO.empty;
+        }
```

Comment:
`return tag.map(converter::convertToDto).orElse(null);`
And remove pls `TagDTO.empty` field. It misleads because it is just a `null ` but not an "empty" Tag.

Reply:
Fixed it